src/instrumentations/grpc-js/wrapGrpcClient.ts

```diff
@@ -0,0 +1,90 @@
+import shimmer from 'shimmer';
+import * as grpcClientUtils from '@opentelemetry/instrumentation-grpc/build/src/grpc-js/clientUtils';
+import { metadataCaptureType } from '@opentelemetry/instrumentation-grpc/build/src/internal-types';
+import {
+  GrpcClientFunc,
+  GrpcEmitter,
+} from '@opentelemetry/instrumentation-grpc/build/src/grpc-js/types';
+import type { Client, Metadata } from '@grpc/grpc-js';
+import { Span } from '@opentelemetry/api';
+import { safeExecute } from '@lumigo/node-core/lib/utils';
+import { concatenatePayload } from './utils';
+
+const wrapCallback = (span: Span, original: GrpcClientFunc, args: unknown[]): void => {
+  // args are the parameters that were passed to GrpcClientFunc.
+  // We parse them to find the callback function and the request payload.
+
+  if (!original.responseStream) {
+    const callbackFuncIndex = args.findIndex((arg) => {
+      return typeof arg === 'function';
+    });
+    if (callbackFuncIndex !== -1) {
+      const originalCallback = args[callbackFuncIndex];
+      args[callbackFuncIndex] = (err, res) => {
+        safeExecute(() => {
+          span.setAttribute('rpc.request.payload', JSON.stringify(args[0]));
+          span.setAttribute('rpc.response.payload', JSON.stringify(res));
+        })();
+        // @ts-ignore - we assume that originalCallback has this signature (the type of args is unfortunately broad)
+        return originalCallback(err, res);
+      };
+    }
+  }
+};
+
+const wrapResponseStream = (
+  span: Span,
+  original: GrpcClientFunc,
+  args: unknown[],
+  call: GrpcEmitter
+) => {
+  if (original.responseStream) {
+    span.setAttribute('rpc.request.payload', JSON.stringify(args[0]));
```

Comment:
don't you always want to set attributes on safeExecute

Reply:
Important question. 
Line 42 is inside safeExecute (line 57).
You see another safeExecute in this function because it defines a callback that will be executed in another context.
Can you be more specific? what exact line do you think is not safe?

Reply:
I was just wondering whether where you set attributes within a safeExecute scope or not - but I think I get it now (whenever inside a callback?)